ComposerChallenge/Views/ComposerView.swift

```diff
@@ -75,39 +105,58 @@ class ComposerView: UIView {
     // I'm leaving instructions to complete in each method to get them working
     
     // Constructing subviews means adding each view to the base view
-    // There is a method Apple proviedes that you can call directly to add subivews to the base view
+    // There is a method Apple provides that you can call directly to add subivews to the base view
     func constructSubviews() {
+        addSubview(composerImageView)
+        addSubview(nameLabel)
+        addSubview(bioLabel)
+        addSubview(alternateImageButton)
         
     }
     
     // This method will list all of the views you created above, allow them to use NSLayoutConstraints, then choose the NSLayoutConstraint you want for each view
     func constructSubviewConstraints() {
         // Create an array of your views
-
+        let views = [composerImageView, nameLabel, bioLabel, alternateImageButton]
         
-        // For each view, turn their property `.translatesAutoresizingMaskIntoConstraints` to be false
+     // TODO: - Can we replace the constraints below with:
```

Comment:
Yea that's the idea. I wouldn't mix actual constraint logic w/ each view though. Have each view deal w/ that in their own block, or it gets a little harder to debug. 

I was thinking you could use your `views` array like this:

`views.forEach { $0.translatesAutoresizingMaskIntoConstraints = false }`